code/ruby/search_synonyms.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'xapian'
+require 'json'
+require_relative 'support'
+
+### Start of example code.
+def search(dbpath, querystring, offset: 0, pagesize: 10)
+  # offset - defines starting point within result set
+  # pagesize - defines number of records to retrieve
+
+  # Open the database we're going to search.
+  db = Xapian::Database.new(dbpath)
+
+  # puts db.methods
+  # Start of adding synonyms
+  # db.add_synonym("time", "calendar")
```

Comment:
You need a `WritableDatabase` for this (which the other versions of this use).

Reply:
Resolved in 96bf0e42bb98cf5ed94517e0f02b9e5675e3f3d4.

Reply:
Thanks!